src/services/homework/hooks/index.js

```diff
@@ -94,127 +110,185 @@ const hasViewPermissionAfter = (hook) => {
-				const gradedSubmissions = currentSubmissions.filter(isGraded);
-				const NumberOfUsersWithSubmission = validSubmissions.map((e) => (e.courseGroupId ? ((e.courseGroupId.userIds || []).length || 1) : ((e.teamMembers || []).length || 1))).reduce((a, b) => a + b, 0);
+				if (
+					!copy.private &&
+					((((copy.courseId || {}).userIds || []).includes(
+						hook.params.account.userId.toString()
+					) &&
+						copy.publicSubmissions) ||
+						isTeacher(hook.params.account.userId, copy))
+				) {
+					const NumberOfCourseMembers = ((copy.courseId || {}).userIds || [])
+						.length;
+					const currentSubmissions = submissions.data.filter((s) =>
+						equalIds(copy._id, s.homeworkId)
+					);
+					const validSubmissions = currentSubmissions.filter(isValidSubmission);
+					const gradedSubmissions = currentSubmissions.filter(isGraded);
+					const NumberOfUsersWithSubmission = validSubmissions
+						.map((e) =>
+							e.courseGroupId
+								? (e.courseGroupId.userIds || []).length || 1
+								: (e.teamMembers || []).length || 1
+						)
+						.reduce((a, b) => a + b, 0);
 
-				const NumberOfGradedUsers = gradedSubmissions.map((e) => (e.courseGroupId ? ((e.courseGroupId.userIds || []).length || 1) : ((e.teamMembers || []).length || 1))).reduce((a, b) => a + b, 0);
-				const submissionPerc = (NumberOfCourseMembers !== 0)
-					? (NumberOfUsersWithSubmission / NumberOfCourseMembers) * 100
-					: 0;
-				const gradePerc = (NumberOfCourseMembers !== 0)
-					? (NumberOfGradedUsers / NumberOfCourseMembers) * 100
-					: 0;
+					const NumberOfGradedUsers = gradedSubmissions
+						.map((e) =>
+							e.courseGroupId
+								? (e.courseGroupId.userIds || []).length || 1
+								: (e.teamMembers || []).length || 1
+						)
+						.reduce((a, b) => a + b, 0);
+					const submissionPerc =
+						NumberOfCourseMembers !== 0
+							? (NumberOfUsersWithSubmission / NumberOfCourseMembers) * 100
+							: 0;
+					const gradePerc =
+						NumberOfCourseMembers !== 0
+							? (NumberOfGradedUsers / NumberOfCourseMembers) * 100
+							: 0;
 
-				copy.stats = {
-					userCount: ((copy.courseId || {}).userIds || []).length,
-					submissionCount: NumberOfUsersWithSubmission,
-					submissionPercentage: (submissionPerc != Infinity) ? submissionPerc.toFixed(2) : undefined,
-					gradeCount: NumberOfGradedUsers,
-					gradePercentage: (gradePerc != Infinity) ? gradePerc.toFixed(2) : undefined,
-					averageGrade: getAverageRating(currentSubmissions),
-				};
-				copy.isTeacher = isTeacher(hook.params.account.userId, copy);
+					copy.stats = {
+						userCount: ((copy.courseId || {}).userIds || []).length,
+						submissionCount: NumberOfUsersWithSubmission,
+						submissionPercentage:
+							submissionPerc != Infinity
+								? submissionPerc.toFixed(2)
+								: undefined,
+						gradeCount: NumberOfGradedUsers,
+						gradePercentage:
+							gradePerc != Infinity ? gradePerc.toFixed(2) : undefined,
```

Comment:
Codacy found an issue: [Expected '!==' and instead saw '!='.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5989783)